File: models/exams.rb
require_relative 'istudiez'

module IStudiez
  class Exam
    include IStudiez::SyncdTbl
    storage_names[:default] = 'exams'

    property :uid, Serial

    property :building,               Text
    property :notes,                  Text
    property :completion_day,         Integer
    property :weight_uid,             Integer
    property :complete,               Boolean
    property :address,                Text
    property :date,                   Integer
    property :time_duration,          Integer
    property :title,                  Text
    property :room,                   Text
    property :earned_points,          Decimal
    property :course_uid,             Integer
    property :total_points,           Decimal
    property :time_offset,            Integer
    property :completion_time_offset, Integer
    property :all_day,                Boolean
    property :url,                    Text
  end
end

File: nodejs/models/exams.js
/* jshint indent: 1 */

module.exports = function(sequelize, DataTypes) {
	return sequelize.define('exams', {
		uid: {
			type: DataTypes.INTEGER,
			allowNull: true,
			primaryKey: true
		},
		is_new: {
			type: "BOOL",
			allowNull: false,
			defaultValue: '0'
		},
		is_local: {
			type: "BOOL",
			allowNull: false,
			defaultValue: '0'
		},
		DEL: {
			type: "BOOL",
			allowNull: false,
			defaultValue: '0'
		},
		server_uid: {
			type: DataTypes.INTEGER,
			allowNull: true
		},
		old_server_uid: {
			type: DataTypes.INTEGER,
			allowNull: true
		},
		building: {
			type: DataTypes.STRING(2000),
			allowNull: true
		},
		all_day: {
			type: "bool",
			allowNull: true
		},
		notes: {
			type: DataTypes.STRING(2000),
			allowNull: true
		},
		completion_day: {
			type: DataTypes.INTEGER,
			allowNull: true
		},
		weight_uid: {
			type: DataTypes.INTEGER,
			allowNull: true
		},
		complete: {
			type: "bool",
			allowNull: true
		},
		address: {
			type: DataTypes.STRING(2000),
			allowNull: true
		},
		date: {
			type: DataTypes.INTEGER,
			allowNull: true
		},
		time_duration: {
			type: DataTypes.INTEGER,
			allowNull: true
		},
		title: {
			type: DataTypes.STRING(2000),
			allowNull: true
		},
		room: {
			type: DataTypes.STRING(2000),
			allowNull: true
		},
		earned_points: {
			type: "double",
			allowNull: true
		},
		course_uid: {
			type: DataTypes.INTEGER,
			allowNull: true
		},
		total_points: {
			type: "double",
			allowNull: true
		},
		time_offset: {
			type: DataTypes.INTEGER,
			allowNull: true
		},
		url: {
			type: DataTypes.STRING(2000),
			allowNull: true
		},
		completion_time_offset: {
			type: DataTypes.INTEGER,
			allowNull: true
		}
	}, {
		tableName: 'exams'
	});
};
